runtime/compiler/compile/J9NonHelperSymbols.enum

```diff
@@ -0,0 +1,70 @@
+/*******************************************************************************
+ * Copyright (c) 2000, 2021 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+/*
+ * This file will be included within an enum.  Only comments, preprocessor macros,
+ * and enumerator definitions are permitted.
+ */
+
+   #include "compile/OMRNonHelperSymbols.enum"
+
+   J9firstNonhelperSymbol,
+
+   classRomPtrSymbol = J9firstNonhelperSymbol,
+   ramStaticsFromClassSymbol,
+   componentClassAsPrimitiveSymbol,
+   initializeStatusFromClassSymbol,
+   threadPrivateFlagsSymbol,  // private flags slot on j9vmthread
+   arrayletSpineFirstElementSymbol,// address of contiguous arraylet spine first element
+   dltBlockSymbol,            // DLT Block field in j9vmthread
+   lowTenureAddressSymbol,    // on j9vmthread
+   highTenureAddressSymbol,   // on j9vmthread
+   fragmentParentSymbol,
+   globalFragmentSymbol,
```

Comment:
aren't the 4 above defined in OMR ?

Reply:
I don't think so...   The `lowTenureAddressSymbol` and `highTenureAddressSymbol` appear once within a `J9_PROJECT_SPECIFIC` block in LoopVersioner, but that's about it.   You're right that that entire project-specific chunk could be refactored at some point though.

Reply:
I think there is a problem with classRomPtr enum.  I'll look into it shortly.  I'll keep this WIP in the meantime.

Reply:
Sorry it took so long to get back to this.  I checked each entry in the enum and it is fine (note that the enum is not "enabled" as a result of this PR, just the J9-specific entries are added to it).  The `classRomPtr` generation function is also migrated (via duplication) into the J9 namespace.  It will be cleaned up in OMR 5822.

Reply:
sorry, I didn't mean the JIT code that accesses those fields. I thought the fields themselves (on j9vmthread or in the GC structure holding the fragment values) were defined in OMR...

Reply:
@0xdaryl aren't the corresponding fields (`lowTenureAddress`, `highTenureAddress`, `fragmentParent`, and `globalFragment`) defined in OMR? In which case it seems a bit strange to be moving them to OpenJ9, unless you think those fields should also be moved to OpenJ9 (at some point?) ...

Reply:
Yes, I understand what you mean now and I removed those 4 fields from the list.  I will add them back to OMR in the follow-on OMR PR.

I think ultimately they may move to OpenJ9, but as they represent fields in data structures that are in OMR and not in the compiler there will be effort required in considering their purpose is and migrating to OpenJ9.  I'll save that for another day.